### PR TITLE
package.yml: various fixes and improvements

### PR DESCRIPTION
**Pull request by Staudey**
## Description

Fixes some small mistakes resp. outdated info, unifies style across sections, adds some additional info, applies `yarn prettier` formatting

### Submitter Checklist

- [x] Squashed commits with `git rebase -i` (if needed)

### CODE REVIEW

File: docs/packaging/package.yml.md [hunk 1]
```diff
@@ -85,34 +86,34 @@ The packaging steps are all considered optional, however the absence of the `ins
...
 | ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | **setup**   | Performed after the source extraction. This is the correct place to perform any `configure` routine, or to `patch` the sources.                            |
-| **build**   | Use this step to run the build portion, i.e. `make`                                                                                                        |
-| **install** | This is where you should install the files into the final packaging directory, i.e. `make install`                                                         |
-| **check**   | This is where tests / checking should occur, i.e. `make check`                                                                                             |
+| **build**   | Use this step to run the build portion, e.g. `make`                                                                                                        |
```
Comment (davidjharder) [inline docs/packaging/package.yml.md:89]: Let's avoid Latin. Please replace "e.g". with "for example", surrounded by commas, throughout this page.

File: docs/packaging/package.yml.md [hunk 2]
```diff
@@ -121,43 +122,43 @@ Macros are prefixed with `%`, and are substituted before your script is executed
...
-| **%ninja_build**     | Runs ninja and passes our `%JOBS%` variable. This macro obsoletes _%meson_build_.                                     |
-| **%ninja_install**   | Runs meson install and passed the appropriate `DESTDIR` and `%JOBS%` variable. This macro obsoletes _%meson_install_. |
-| **%ninja_check**     | Runs ninja test and passes our `%JOBS%` variable. This macro obsoletes _%meson_check_.                                |
+| Macro                | Description                                                                                                             |
+| -------------------- | ----------------------------------------------------------------------------------------------------------------------- |
+| **%meson_configure** | Runs `meson` with our CFLAGS and appropriate flags such as `libdir`.                                                    |
+| **%ninja_build**     | Runs `ninja` and passes our `%JOBS%` variable. This macro obsoletes _%meson_build_.                                     |
+| **%ninja_install**   | Runs `ninja install` and passed the appropriate `DESTDIR` and `%JOBS%` variable. This macro obsoletes _%meson_install_. |
```
Comment (davidjharder) [inline docs/packaging/package.yml.md:152]: "and passed" -> "and passes"

File: docs/packaging/package.yml.md [hunk 3]
```diff
@@ -174,26 +175,27 @@ Macros are prefixed with `%`, and are substituted before your script is executed
...
-| Macro              | Description                                                                 |
-| ------------------ | --------------------------------------------------------------------------- |
-| **%waf_configure** | Runs waf configure with prefix.                                             |
-| **%waf_build**     | Runs waf and passes our `%JOBS%` variable.                                  |
-| **%waf_install**   | Runs waf install and passed the appropriate `destdir` and `%JOBS%` variable |
+| Macro              | Description                                                                   |
+| ------------------ | ----------------------------------------------------------------------------- |
+| **%waf_configure** | Runs `waf configure` with prefix.                                             |
+| **%waf_build**     | Runs `waf` and passes our `%JOBS%` variable.                                  |
+| **%waf_install**   | Runs `waf install` and passed the appropriate `destdir` and `%JOBS%` variable |
```
Comment (davidjharder) [inline docs/packaging/package.yml.md:198]: "and passed" -> "and passes"

**Review by davidjharder — CHANGES_REQUESTED** [review]
Thanks for the fine-toothed review. I found more nits :)

Can you please also standardize on ending all lines in the table with a period. We are not consistent here I noticed.

File: docs/packaging/package.yml.md [hunk 4]
```diff
@@ -85,34 +86,34 @@ The packaging steps are all considered optional, however the absence of the `ins
...
-| **thin-lto**     | Enable Thin Link Time Optimization `-flto=thin` with a supported linker.           |
-| **lto**          | Enable Link Time Optimization `-flto`.                                             |
+| Optimize Value   | Description                                                                         |
+| ---------------- | ----------------------------------------------------------------------------------- |
+| **speed**        | Optimises the package for performance `-O3` plus other flags.                       |
+| **size**         | Optimises the package build to minimize size `-Os`. Not supported with clang.       |
+| **no-bind-now**  | Configures the package to disable certain flags, where RELRO is unsupported.        |
+| **no-symbolic**  | Disables `-Wl,-Bsymbolic-functions` linker flag.                                    |
+| **unroll-loops** | Enables `-funroll-loops`. Use this sparingly, only when it provides proven benefit. |
+| **runpath**      | Enablse `-Wl,--enable-new-dtags` to make linker use RUNPATH's instead of RPATH's.   |
```
[the remark this answers is not in the document]
Comment (Staudey) [reply docs/packaging/package.yml.md]: lol, done

**Comment by Staudey** [comment]
I think I addressed everything. In any case I'll have to postpone additional work until tomorrow because I can't keep my eyes open much longer ^^

**Review by davidjharder — APPROVED** [review]
Not bad for 3AM work :)
Thanks Staudey!

**Pull request merged**